[FL-2529][FL-1628] New LF-RFID subsystem

DrZlo13 opened the pull request:
# What's new

New protocols:
- Indala26 (covers almost all Indala PSK cards)
- AWID
- FDX-A
- FDX-B
- 96-bit HID protocol (covers up to HID 40-bit)
- 192-bit HID protocol (covers HID extended)
- RAW RFID data recording

# Verification 

- Use the LF-RFID application as usual, try to read, emulate, write new protocols.

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

DrZlo13 committed:
Makefile: unit tests pack

DrZlo13 committed:
RFID: pulse joiner and its unit test

DrZlo13 committed:
Move pulse protocol helpers to appropriate place

DrZlo13 committed:
Drop pulse_joiner tests

DrZlo13 committed:
Generic protocol, protocols dictionary, unit test

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
Protocol dict unit test

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
iButton: protocols dictionary

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
Lib: varint

DrZlo13 committed:
Lib: profiler

DrZlo13 committed:
Unit test: varint

DrZlo13 committed:
rfid: worker mockup

DrZlo13 committed:
LFRFID: em4100 unit test

DrZlo13 committed:
Storage: file_exist function

DrZlo13 committed:
rfid: fsk osc

DrZlo13 committed:
rfid: generic fsk demodulator

DrZlo13 committed:
rfid: protocol em4100

DrZlo13 committed:
rfid: protocol h10301

DrZlo13 committed:
rfid: protocol io prox xsf

DrZlo13 committed:
Unit test: rfid protocols

DrZlo13 committed:
rfid: new hal

DrZlo13 committed:
rfid: raw worker

DrZlo13 committed:
Unit test: fix error output

DrZlo13 committed:
rfid: worker

DrZlo13 committed:
rfid: plain c cli

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
fw: migrate to scons

DrZlo13 committed:
lfrfid: full io prox support

DrZlo13 committed:
unit test: io prox protocol

DrZlo13 committed:
SubGHZ: move bit defines to source

DrZlo13 committed:
FSK oscillator: level duration compability

DrZlo13 committed:
libs: bit manipulation library

DrZlo13 committed:
lfrfid: ioprox protocol, use bit library and new level duration method of FSK ocillator

DrZlo13 committed:
bit lib: unit tests

DrZlo13 committed:
Bit lib: parity tests, remove every nth bit, copy bits

DrZlo13 committed:
Lfrfid: awid protocol

DrZlo13 committed:
bit lib: uint16 and uint32 getters, unit tests

DrZlo13 committed:
lfrfid: FDX-B read, draft version

DrZlo13 committed:
Minunit: better memeq assert

DrZlo13 committed:
bit lib: reverse, print, print regions

DrZlo13 committed:
Protocol dict: get protocol features, get protocol validate count

DrZlo13 committed:
lfrfid worker: improved read

DrZlo13 committed:
lfrfid raw worker: psk support

DrZlo13 committed:
Cli: rfid plain C cli

DrZlo13 committed:
protocol AWID: render

DrZlo13 committed:
protocol em4100: render

DrZlo13 committed:
protocol h10301: render

DrZlo13 committed:
protocol indala26: support every indala 26 scramble

DrZlo13 committed:
Protocol IO Prox: render

DrZlo13 committed:
Protocol FDX-B: advanced read

DrZlo13 committed:
lfrfid: remove unused test fuction

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
lfrfid: fix os primitives

DrZlo13 committed:
bit lib: crc16 and unit tests

DrZlo13 committed:
FDX-B: save data

DrZlo13 committed:
lfrfid worker: increase stream size. Alloc raw worker only when needed.

DrZlo13 committed:
lfrfid: indala26 emulation

DrZlo13 committed:
lfrfid: prepare to write

DrZlo13 committed:
lfrfid: fdx-b emulation

DrZlo13 committed:
lfrfid: awid, ioprox write

DrZlo13 committed:
lfrfid: write t55xx w\o validation

DrZlo13 committed:
lfrfid: better t55xx block0 handling

DrZlo13 committed:
lfrfid: use new t5577 functions in worker

DrZlo13 committed:
lfrfid: improve protocol description

DrZlo13 committed:
lfrfid: write and verify

DrZlo13 committed:
lfrfid: delete cpp cli

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
lfrfid: improve worker usage

DrZlo13 committed:
lfrfid-app: step to new worker

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
lfrfid: old indala (I40134) load fallback

DrZlo13 committed:
lfrfid: indala26, recover wrong synced data

DrZlo13 committed:
lfrfid: remove old worker

DrZlo13 committed:
lfrfid app: dummy read screen

DrZlo13 committed:
lfrfid app: less dummy read screen

hedger commented on applications/archive/helpers/archive_apps.c:
```diff
@@ -29,23 +29,13 @@ bool archive_app_is_available(void* context, const char* path) {
-            }
+        Storage* storage = furi_record_open(RECORD_STORAGE);
+
+        if(storage_file_exists(storage, "/any/u2f/key.u2f")) {
```
ANY_PATH("u2f/key.u2f")

hedger commented on applications/lfrfid/lfrfid_app.cpp:
```diff
@@ -37,15 +37,35 @@ LfRfidApp::LfRfidApp()
     , storage{"storage"}
```
RECORD_STORAGE

hedger commented on applications/lfrfid/lfrfid_app.cpp:
```diff
@@ -37,15 +37,35 @@ LfRfidApp::LfRfidApp()
     , storage{"storage"}
     , dialogs{"dialogs"}
```
RECORD_DIALOGS

hedger commented on applications/lfrfid/lfrfid_cli.c:
```diff
@@ -0,0 +1,643 @@
+
+// app cli function
+void lfrfid_on_system_start() {
+    Cli* cli = furi_record_open("cli");
```
RECORD_CLI

hedger commented on applications/lfrfid/lfrfid_cli.c:
```diff
@@ -0,0 +1,643 @@
+void lfrfid_on_system_start() {
+    Cli* cli = furi_record_open("cli");
+    cli_add_command(cli, "rfid", CliCommandFlagDefault, lfrfid_cli, NULL);
+    furi_record_close("cli");
```
RECORD_CLI

hedger commented on applications/lfrfid/lfrfid_cli.c:
```diff
@@ -0,0 +1,643 @@
+    stream_free(stream);
+    string_clear(filepath);
+    string_clear(info_string);
+    furi_record_close("storage");
```
RECORD_STORAGE

hedger commented on applications/lfrfid/lfrfid_cli.c:
```diff
@@ -0,0 +1,643 @@
+    string_t filepath, info_string;
+    string_init(filepath);
+    string_init(info_string);
+    Storage* storage = furi_record_open("storage");
```
RECORD_STORAGE

hedger commented on applications/lfrfid/lfrfid_cli.c:
```diff
@@ -0,0 +1,643 @@
+    protocol_dict_free(dict);
+}
+
+static void lfrfid_cli_raw_analyze(Cli* cli, string_t args) {
```
150+ lines function

DrZlo13 replied:
Was simplified

hedger commented on applications/lfrfid/lfrfid_cli.c:
```diff
@@ -0,0 +1,643 @@
+
+    string_t filepath;
+    string_init(filepath);
+    Storage* storage = furi_record_open("storage");
```
RECORD_STORAGE

hedger commented on applications/lfrfid/lfrfid_cli.c:
```diff
@@ -0,0 +1,643 @@
+
+    } while(false);
+
+    furi_record_close("storage");
```
RECORD_STORAGE

hedger commented on applications/lfrfid/scene/lfrfid_app_scene_read_success.cpp:
```diff
@@ -4,10 +4,37 @@
+            string_cat_printf(string_info, "...");
+            break;
+        } else {
+            string_cat_printf(string_info, "%02X", data[i]);
```
move to a helper function?

hedger commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,602 @@
+bool lfrfid_raw_worker_start_emulate(LFRFIDRawWorker* worker, const char* file_path) {
+    furi_check(furi_thread_get_state(worker->thread) == FuriThreadStateStopped);
+
+    bool result = true;
```
unused

hedger commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,602 @@
+}
+
+bool lfrfid_raw_worker_stop(LFRFIDRawWorker* worker) {
+    bool result = true;
```
unused

hedger commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,602 @@
+        }
+
+        if(!ctx->buffers[id].occupied) {
+            memcpy(
```
move block to separate function?

DrZlo13 replied:
Moved

hedger commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,602 @@
+static int32_t lfrfid_raw_read_worker_thread(void* thread_context) {
+    LFRFIDRawWorker* worker = (LFRFIDRawWorker*)thread_context;
+
+    Storage* storage = furi_record_open("storage");
```
RECORD_STORAGE

hedger commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,602 @@
+
+    vStreamBufferDelete(data->ctx.stream);
+    stream_free(stream);
+    furi_record_close("storage");
```
RECORD_STORAGE

hedger commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,602 @@
+
+    if(half) {
+        uint32_t flag = HalfTransfer;
+        size_t len = xStreamBufferSendFromISR(ctx->stream, &flag, sizeof(uint32_t), pdFALSE);
```
copypasted

hedger replied:
move outside?

hedger commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,602 @@
+
+    LFRFIDRawWorkerEmulateData* data = malloc(sizeof(LFRFIDRawWorkerEmulateData));
+
+    Storage* storage = furi_record_open("storage");
```
RECORD_STORAGE

hedger commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,602 @@
+
+                buffer_p->size = 0;
+                __DMB();
+                buffer_p->occupied = false;
```
volatile?

DrZlo13 replied:
volatile will load the cpu more than we can afford, this place is extremely sensitive to that

hedger commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,602 @@
+        free(header_text);
+
+        uint32_t max_buffer_size;
+        size = stream_read(data->stream, (uint8_t*)&max_buffer_size, sizeof(uint32_t));
```
same code in CLI loader, move to 1 place?

DrZlo13 replied:
Moved

hedger commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,602 @@
+
+    stream_free(data->stream);
+    vStreamBufferDelete(data->ctx.stream);
+    furi_record_close("storage");
```
RECORD_STORAGE

hedger commented on lib/lfrfid/lfrfid_worker_modes.c:
```diff
@@ -0,0 +1,514 @@
+        furi_hal_rfid_tim_read(125000, 0.5);
+        FURI_LOG_D(TAG, "Start ASK");
+    } else {
+        furi_hal_rfid_tim_read(62500, 0.25);
```
magic

DrZlo13 replied:
Literally frequency and duty cycle, as furi_hal_rfid_tim_read arguments says. I don't see the point in naming it.

hedger commented on lib/lfrfid/lfrfid_worker_modes.c:
```diff
@@ -0,0 +1,514 @@
+
+    if(half) {
+        uint32_t flag = HalfTransfer;
+        xStreamBufferSendFromISR(stream, &flag, sizeof(uint32_t), pdFALSE);
```
move out of if()

hedger commented on lib/lfrfid/lfrfid_worker_modes.c:
```diff
@@ -0,0 +1,514 @@
+
+    switch(worker->read_type) {
+    case LFRFIDWorkerReadTypePSKOnly:
+        lfrfid_raw_worker_start_read(raw_worker, worker->raw_filename, 62500, 0.25);
```
magic #2

DrZlo13 replied:
Literally frequency and duty cycle, as furi_hal_rfid_tim_read arguments says. I don't see the point in naming it.

hedger commented on lib/lfrfid/protocols/lfrfid_protocols.h:
```diff
@@ -0,0 +1,32 @@
+
+typedef struct {
+    LFRFIDWriteType write_type;
+    union {
```
union?

DrZlo13 replied:
To make it easier to extend the code later. We want not only T5577.

hedger commented on lib/lfrfid/protocols/protocol_awid.c:
```diff
@@ -0,0 +1,239 @@
+
+const ProtocolBase protocol_awid = {
+    .name = "AWID",
+    .manufacturer = "AWIG",
```
awid?

DrZlo13 replied:
Applied Wireless Identifications Data by Applied Wireless Identifications Group, iirc

hedger commented on lib/lfrfid/protocols/protocol_h10301.c:
```diff
@@ -0,0 +1,386 @@
+    protocol_h10301_write_raw_bit(1, 7, card_data);
+
+    // company / OEM code 1
+    protocol_h10301_write_bit(0, 8, card_data);
```
maybe use a cycle?

DrZlo13 replied:
h10301 needs rework, it will be done later

hedger commented on lib/lfrfid/protocols/protocol_io_prox_xsf.c:
```diff
@@ -0,0 +1,297 @@
+    }
+
+    // ... check for known ones...
+    if(bit_lib_bit_is_not_set(encoded_data[2], 6)) {
```
use cycle?

DrZlo13 replied:
will look bad

hedger commented on lib/toolbox/protocols/protocol_dict.c:
```diff
@@ -0,0 +1,226 @@
+        if(features & feature) {
+            ProtocolDecoderFeed fn = dict->base[i]->decoder.feed;
+
+            if(fn) {
```
move to single condition

DrZlo13 replied:
will be less readable

hedger commented on lib/toolbox/pulse_joiner.c:
```diff
@@ -0,0 +1,117 @@
+    furi_check((pulse_joiner->pulse_index + 1) < PULSE_MAX_COUNT);
+
+    if(polarity == false && pulse_joiner->pulse_index == 0) {
+        // first negative pulse is ommited
```
omitted

hedger commented on lib/toolbox/varint.h:
```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <stdint.h>
+#include <stdlib.h>
+
```
c++ header wrapper?

hedger commented on lib/lfrfid/tools/fsk_demod.h:
```diff
@@ -0,0 +1,12 @@
+#include <stdbool.h>
+
+typedef struct FSKDemod FSKDemod;
+
```
c++ header wrapper?

hedger commented on lib/toolbox/profiler.h:
```diff
@@ -0,0 +1,15 @@
+#pragma once
+
```
c++ header wrapper?

hedger commented on lib/toolbox/pulse_joiner.h:
```diff
@@ -0,0 +1,38 @@
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+
```
c++ header wrapper?

hedger reviewed (changes requested):
Remove .vscode.zip

DrZlo13 committed:
lfrfid: generic 96-bit HID protocol (covers up to HID 37-bit)

DrZlo13 committed:
rename

DrZlo13 committed:
lfrfid: improve indala26 read

DrZlo13 committed:
lfrfid: generic 192-bit HID protocol (covers all HID extended)

DrZlo13 committed:
lfrfid: TODO about HID render

DrZlo13 committed:
lfrfid: new protocol FDX-A

DrZlo13 committed:
lfrfid-app: correct worker stop on exit

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
misc fixes

DrZlo13 committed:
lfrfid: FDX-A and HID distinguishability has been fixed.

ide committed:
lfrfid: decode HID size header and render it (#1612)

lfrfid: decode HID size header and render it (#1612)

DrZlo13 committed:
lfrfid: rename HID96 and HID192 to HIDProx and HIDExt

DrZlo13 committed:
lfrfid: extra actions scene

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

ide committed:
lfrfid: decode generic HID Proximity size lazily (#1618)

This fixes a bug when displaying info about a captured HID key on the Flipper. Previously, `protocol->protocol_size` was set when decoding raw bits. However, the size was not set when loading a key that was already decoded and saved to disk (or a manually entered key).

This commit lazily computes the key size with a new function named `protocol_hid_generic_decode_protocol_size` and removes the `protocol_size` from the HID protocol struct.

Tested this fix by loading a saved HID key on the Flipper and verifying that the Flipper shows the key size.

DrZlo13 committed:
lib: stream of data buffers concept

DrZlo13 committed:
lfrfid: raw file helper

DrZlo13 committed:
lfrfid: changed raw worker api

DrZlo13 committed:
lfrfid: packed varint pair

DrZlo13 committed:
lfrfid: read stream speedup

DrZlo13 committed:
lfrfid app: show read mode

DrZlo13 committed:
Documentation

exussum12 commented:
lib/littlefs/lfs.c:1070: undefined reference to `furi_log_print'

I'm getting that when compiling

hedger commented on lib/lfrfid/lfrfid_raw_file.c:
```diff
@@ -0,0 +1,149 @@
+        .max_buffer_size = max_buffer_size};
+
+    size_t size = stream_write(file->stream, (uint8_t*)&header, sizeof(LFRFIDRawFileHeader));
+    if(size == sizeof(LFRFIDRawFileHeader)) {
```
return (size == sizeof(LFRFIDRawFileHeader));

hedger commented on lib/lfrfid/lfrfid_raw_file.c:
```diff
@@ -0,0 +1,149 @@
+    size = stream_write(file->stream, buffer_data, buffer_size);
+    if(size != buffer_size) return false;
+
+    return true;
```
return (stream_write() == sizeof()) && (stream_write() == buffer_size)
would be easier to read. Personal opinion, completely optional

hedger commented on lib/lfrfid/protocols/protocol_fdx_a.c:
```diff
@@ -0,0 +1,243 @@
+        decoded_data[i] &= 0x7F;
+    }
+
+    if(parity_sum == 0) {
```
return (parity_sum == 0)

hedger commented on lib/toolbox/buffer_stream.c:
```diff
@@ -0,0 +1,145 @@
+    if(!buffer_write(buffer, data, size)) {
+        // if buffer is full - send it
+        buffer->occupied = true;
+        // we always has space for buffer in stream
```
have

hedger commented:
@exussum12 either use pre-built verison from our CI in comments above, or run fbt with -c flag

exussum12 commented on lib/lfrfid/lfrfid_dict_file.c:
```diff
@@ -0,0 +1,181 @@
+
+        if(!flipper_format_write_hex(file, "Data", data, data_size)) break;
+        result = true;
+    } while(0);
```
Minor but while(0) used here, while (false) used on 173, should likely be the same

DrZlo13 committed:
lfrfid app: raw read gui

DrZlo13 committed:
lfrfid app: storage check for raw read

DrZlo13 committed:
memleak fix

DrZlo13 committed:
lint

DrZlo13 committed:
review fixes

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

xMasterX commented on SConstruct:
```diff
@@ -168,6 +168,8 @@ copro_dist = distenv.CoproBuilder(
 distenv.Alias("copro_dist", copro_dist)
 
 firmware_flash = distenv.AddOpenOCDFlashTarget(firmware_env)
+if distenv["FORCE"]:
```
Looks like some old code left that has been removed/changed in dev

(Same thing for 2-3 files below)

See commit:
[SConstruct](https://github.com/flipperdevices/flipperzero-firmware/commit/a1637e9216e98396938debd7ccb4e15cbcb41eb1#diff-d78c5c377c03dd251edc72cdb0ae7d9986fa08361eb392187218f8c670c0bc4e)

DrZlo13 replied:
Good catch.

DrZlo13 committed:
lfrfid app: read blink color

DrZlo13 committed:
lfrfid app: reset key name after read

DrZlo13 committed:
review fixes

wosk commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,357 @@
+            uint32_t flags = furi_event_flag_get(worker->events);
+            if(FURI_BIT(flags, LFRFIDRawWorkerEventStop)) {
+                break;
+            };
```
```suggestion
            }
```

wosk commented on lib/lfrfid/lfrfid_raw_worker.c:
```diff
@@ -0,0 +1,357 @@
+
+            if(FURI_BIT(flags, LFRFIDRawWorkerEventStop)) {
+                break;
+            };
```
```suggestion
            }
```

DrZlo13 committed:
lfrfid app: fix copypasted text

DrZlo13 committed:
review fixes

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

kubrickfr commented:
I tried this branch.

It seems like neither of the fobs I own support the protocols implemented here. 

But, before this PR, the device LED was flashing yellow when detecting a device, now there is no visual clue at all that at least "something" is working.

DrZlo13 committed:
lfrfid: disable debug gpio

DrZlo13 committed:
lfrfid: card detection events

kubrickfr commented:
> But, before this PR, the device LED was flashing yellow when detecting a device, now there is no visual clue at all that at least "something" is working.

d8b8f74 fixed it. Thank you @DrZlo13

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
lfrfid: change validation color from magenta to green

DrZlo13 committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
Update core_defines.h

DrZlo13 committed:
lfrfid: prefix fdx-b id by zeroes

ide committed:
lfrfid: parse up to 43-bit HID Proximity keys (#1640)

There appear to be extended HID Proximity keys that are longer than 37 bits that fit into a 44-bit payload (88 raw bits after Manchester encoding, plus the 0x1D preamble for a total of 96 raw bits). These keys don't start with 0b100111 like some extended HID keys supposedly do.

One way to detect these keys is to see if any of the first six bits in the 44-bit payload is a 1. If so, the key is longer than 37 bits. This means the longest key that can fit into a 44-bit payload is 43 bits (the first bit is a 1, followed by 43 key bits).

Also fixed a bug rendering the first nibble of a HID key in `protocol_hid_generic_string_cat_protocol_bits`.

Tested by running `raw_analyze` on known RFID captures:

```
rfid raw_analyze /ext/hid40.raw
...
   Frequency: 125000.000000
  Duty Cycle: 0.500000
       Warns: 0
   Pulse sum: 25415
Duration sum: 40179
     Average: 0.632544
    Protocol: HIDProx [1X XX XX XX XX X0]
 Description:
40-bit HID Proximity
Data: XXXXXXXXXX
```
```
rfid raw_analyze /ext/hid36.raw
...
   Frequency: 125000.000000
  Duty Cycle: 0.500000
       Warns: 0
   Pulse sum: 28080
Duration sum: 41618
     Average: 0.674708
    Protocol: HIDProx [03 XX XX XX XX X0]
 Description:
36-bit HID Proximity
Data: XXXXXXXXX
```

skotopes committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

skotopes committed:
Fbt: downgrade toolchain and fix PS1

Tozapid commented:
Unfortunately not working on my pets. They have  ISO Transponder Microchip 2.12*12mm, think this one - https://www.animal-microchip.com/sale-9568205-professional-iso-transponder-microchip-2-12-12mm-for-animals-identification.html


R/W STANDARD: | ISO 11784/5 FDX-B 134.2khz
-- | --

But with your branch there is no reaction at all. On version 0.64.3, I can see the LED indication that the chip is responding


Upd. My mistake, I had to choose Extra Actions in the menu. So, now I see that LED is blinking green(or yellow, not sure), but still not reading the microchip 

Upd2. Found the spot:) Thank you! It is works
![Screenshot-20220823-155700](https://user-images.githubusercontent.com/9052342/186163637-ed5a6831-8d13-44cf-8ea7-dee1ec80d2e3.png)

DrZlo13 committed:
lfrfid: fix unit tests

DrZlo13 committed:
lfrfid: fix unit tests

skotopes committed:
Merge branch 'dev' into zlo/lf-rfid-overhaul

DrZlo13 committed:
lfrfid app: remove printf

DrZlo13 committed:
lfrfid: indala26, use bit 55 as data

DrZlo13 committed:
lfrfid: indala26, better brief format

DrZlo13 committed:
lfrfid: indala26, loading fallback

DrZlo13 committed:
lfrfid: read timing tuning
